packages/connect-popup/src/index.tsx

```diff
@@ -324,6 +324,20 @@ const handleMessageInCoreMode = (
     handleUIAffectingMessage(message);
 };
 
+const handleWindowBeforeUnload = (_e: BeforeUnloadEvent) => {
+    if (getState().core) {
+        const core = ensureCore();
+        core.handleMessage({
+            type: POPUP.CLOSED,
+            payload: null,
+        });
+    }
+};
+
+const handleParentClosed = () => {
+    window.close();
+};
+
```

Comment:
this is new right? not moved from anywhere?

Reply:
note: related to this https://github.com/trezor/trezor-suite/pull/12198/files

Reply:
Yes, I needed to solve the same issue back then